### DISCUSSION TITLE
Zoom with Command Key and Scroll Wheel Not Working as Expected

### DISCUSSION THREAD

**Discussion by bfzli** [xyflow/xyflow#4896] [General]
The documentation states that 'zoomActivationKeyCode' defaults to Meta, which corresponds to the Command key on Mac. However, scrolling with the wheel to zoom doesn't work as expected; instead, it requires Control + Scroll Wheel. What's wrong here? I want to zoom using the Command key and the scroll wheel. Why is this so difficult to achieve?

```
<ReactFlow
    panOnScroll={true}
    onlyRenderVisibleElements={false}
    zoomOnDoubleClick={false}
    zoomOnScroll={true}
    zoomOnPinch={true}
    draggable={false}
    panOnDrag={true}
    snapToGrid={true}
    preventScrolling={false}
    nodesDraggable={IsPreviewOnly ? false : true}
    panOnScrollSpeed={1}
    ref={ReactflowRef}
    style={styles}
    nodes={Nodes?.length > 0 ? Nodes : NoNodes}
    edges={Edges}
    nodeTypes={nodeTypes}
    onMove={OnPlaygroundMove}
    proOptions={{ hideAttribution: true }}
    maxZoom={ZOOM_LEVELS.MAX}
    minZoom={ZOOM_LEVELS.MIN}
    onPaneClick={OnPanClick}
    selectionOnDrag={true}
    zoomActivationKeyCode='Meta'
    onInit={(r) => OnPlaygroundInit(setReactFlowInstance, r)}
    onEdgeClick={(e) => {
        e.preventDefault()
        e.stopPropagation()
    }}
    onNodeContextMenu={(e, n) => OnNodeContextMenu(
        e, 
        n,
        ReactflowRef
    )}
/>
```

**Comment by bcakmakoglu** [comment]
Your configuration doesn't work.
If you want to zoom with only the zoom key activated, disable `zoomOnScroll` and `zoomOnPinch`.

[Here's an example that works.](https://codesandbox.io/p/sandbox/kind-borg-8j9fyz?file=%2Fsrc%2FApp.tsx%3A38%2C25)

**Reply by bfzli** [reply]
But then this kills the touchpad zoom on laptops, so its fixed one the other feature breaks. Also is there a way to have multiple keys in zoomActivationKeyCode?

**Reply by bfzli** [reply]
The thing is, it scrolls with the wheel on the current configuration. Only the key change is not working because it works with Control and the wheel. I just want to change it to Meta keys, and that’s all.

**Reply by bcakmakoglu** [reply]
Control + wheel causes a *pinch* zoom, not a wheel zoom, hence why you'd have to disable it if you want to *only* allow zooming via a key press.

> Also is there a way to have multiple keys in zoomActivationKeyCode

You can use an array of keys, yes, for example `zoomActivationKeyCode={['Meta', 'Control', 'Shift']}`

**Reply by bfzli** [reply]
I disabled `zoomOnScroll` and `zoomOnPinch` so I could use the zoom keys, but it still isn't working. Neither the touchpad nor the mouse allow zooming in or out by pressing both the wheel and the key.

Here is the code:

```
 return (
        <ReactFlow
            panOnScroll={true}
            onlyRenderVisibleElements={false}
            zoomOnDoubleClick={false}
            zoomOnScroll={false} // true
            zoomOnPinch={false} // true
            draggable={false}
            panOnDrag={true}
            snapToGrid={true}
            preventScrolling={false}
            nodesDraggable={IsPreviewOnly ? false : true}
            panOnScrollSpeed={1}
            ref={ReactflowRef}
            style={styles}
            nodes={Nodes?.length > 0 ? Nodes : NoNodes}
            edges={Edges}
            // @ts-ignore
            nodeTypes={nodeTypes}
            onMove={OnPlaygroundMove}
            proOptions={{ hideAttribution: true }}
            maxZoom={ZOOM_LEVELS.MAX}
            minZoom={ZOOM_LEVELS.MIN}
            onPaneClick={OnPanClick}
            zoomActivationKeyCode={['Meta', 'Control', 'Shift']}
            // @ts-ignore
            onInit={(r) => OnPlaygroundInit(setReactFlowInstance, r)}
            onEdgeClick={(e) => {
                e.preventDefault()
                e.stopPropagation()
            }}
            onNodeContextMenu={(e, n) => OnNodeContextMenu(
                e, 
                n,
                ReactflowRef
            )}
        >
            <Background 
                color='#BDBDBD' 
                variant={BackgroundVariant.Dots} 
                size={2.5}
                gap={24}
            />
        </ReactFlow>
    )
   ```

**Reply by bcakmakoglu** [reply]
Works fine for me in a sandbox on a mac.

First is with trackpad, second with mouse wheel.

https://github.com/user-attachments/assets/88451e5c-a9ae-43e8-90f1-086df8d7fb00

https://github.com/user-attachments/assets/2cf50ce6-febe-4480-bc5f-e26657dfd359

**Reply by bfzli** [reply]
Can you check this with a touchpad? It's not working with my touchpad, and I don't have a trackpad to test it.

**Reply by bcakmakoglu** [reply]
I meant my mac touchpad ^^

**Reply by bcakmakoglu** [reply]
Maybe we should clarify what exactly you're trying to implement here - when *exactly* is zooming supposed to happen?
What gestures etc. are you trying to use? in what combination?
